### PR TITLE
feat: docked, hideName, hideAvatar, css improvements, dockStartOpen

### PR DESCRIPTION
**Pull request by hubertusputu1**
### Features

- Docked / Undocked
- Hide Name
- Hide Avatar
- opened or closed chat window on first time appearance
- custom class name to override the root's css

### Fixes
- CSS improvements

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -22,17 +25,22 @@ import { Chat } from "@empirica/chat";
...
 
 ### Multiple chat instances within the same scope
 
-You can pass an optional `customKey` string prop to differenciate different chats
+You can pass an optional `customKey` string prop to differentiate different chats
+
 within the same scope. This changes which get/set key on the given scope the chat will
-be recorded to.
+
+be recorded.
```
Comment (npaton) [inline Readme.md]: I am not sure why all the blank lines. I don't think it makes sense. https://github.com/empiricaly/chat/blob/de643917dc25cb89bf5776dc2a47904da30c6ffc/Readme.md
Maybe it's just something with your editor.

File: Readme.md [hunk 2]
```diff
@@ -95,34 +111,99 @@ don't like pizza:
...
 
 All components receive the `player`, `scope`, and `customKey` props. `header` and
+
 `closed` also receive an `onClick` prop, that will toggle chat window open and
+
 closed. And `footer` receives `onNewMessage` which new messages should be sent
+
 to. See existing components in `./src` for details.
 
+### Hide the Avatar or Name inside the Message
```
Comment (npaton) [inline Readme.md]:
```suggestion
### Hide the Avatar or the Name
```

File: src/Messages.js [hunk 3]
```diff
@@ -55,10 +57,9 @@ class Messages extends React.PureComponent {
 
 Messages.propTypes = {
   player: PropTypes.object,
-  scope: PropTypes.object.isRequired,
-  customKey: PropTypes.string.isRequired,
   messageComp: PropTypes.elementType,
   filter: PropTypes.func,
+  variant: PropTypes.oneOf(["normal", "name", "avatar"]),
```
Comment (npaton) [inline src/Messages.js]: It looks like we're not using variant anymore.

**Review by npaton — APPROVED** [review]
Looking good, just a few small changes.

For this release we will need to make it a major release, the changes introduced are breaking. ⛵

**Comment by hubertusputu1** [comment]
@npaton Thank you for the review. Yes sure, we will release version 1.2.0 after this PR.

**Comment by npaton** [comment]
Thank you Putu, looking good. Please release a *major* new version, so v2.0.0. This will ensure people who have automatic minor version updating with npm (that is, everyone) do not get broken by the changes here. https://semver.org/ 🚃

**Review by npaton — APPROVED** [review]

**Pull request merged**